sentinelsat_qgis_script.py

```diff
@@ -1,223 +1,307 @@
+
+        def __init__(self, progress):
+            super(self.__class__, self).__init__()
+            self.progress = progress
+
+        def emit(self, record):
+            msg = self.format(record)
             try:
-                products[productid] = api.get_product_odata(productid)
-            except SentinelAPIError as e:
-                if 'Invalid key' in e.msg:
-                    logger.error('No product with ID \'%s\' exists on server', productid)
-    elif name is not None:
-        search_kwargs["identifier"] = name
-        products = api.query(order_by=order_by, limit=limit, **search_kwargs)
-    else:
-        start = start or "19000101"
-        end = end or "NOW"
-        products = api.query(date=(start, end),
-                             order_by=order_by, limit=limit, **search_kwargs)
-
-    if footprints is True:
-        footprints_geojson = api.to_geojson(products)
-        footprints_file = os.path.join(path, "search_footprints.geojson")
-        with open(footprints_file, "w") as outfile:
-            outfile.write(gj.dumps(footprints_geojson))
-        returns['footprints_file'] = footprints_file
-
-    if download is True:
-        product_infos, failed_downloads = api.download_all(products, path)
-        if len(failed_downloads) > 0:
-            with open(os.path.join(path, "corrupt_scenes.txt"), "w") as outfile:
-                for failed_id in failed_downloads:
-                    outfile.write("%s : %s\n" % (failed_id, products[failed_id]['title']))
-    else:
-        for product_id, props in products.items():
+                self.feedback.pushConsoleInfo(msg)
+            except RuntimeError:
+                pass  # no logging
+
+
+    # class ProgressBar(object): TODO update with new progressbar()
+
+    #     def __init__(self, total, initial=0.0, *args, **kwargs):
+    #         self.qgis_progress = _PROGRESS
+    #         self.value = initial
+    #         self.total = total
+    #         self.qgis_progress.setPercentage(self._get_percent())
+
+    #     def _get_percent(self):
+    #         return float(self.value) / self.total * 100
+
+    #     def update(self, increment):
+    #         self.value += increment
+    #         self.qgis_progress.setPercentage(self._get_percent())
+
+    #     def close(self):
+    #         pass
+    class ProgressBar(object):
+        def __init__(self, *args, **kwargs):
+            pass
+        def _get_percent(self):
+            pass
+        def update(self, increment):
+            pass
+        def close(self):
+            pass
+
+    def _set_logger_handler(qgis_progress, level='INFO'):
+        # global logger_set
+        # if logger_set:
+        #     return
+        logger.setLevel(level)
+        h = ProgressHandler(qgis_progress)
+        h.setLevel(level)
+        fmt = logging.Formatter('%(message)s')
+        h.setFormatter(fmt)
+        logger.addHandler(h)
+        logger_set = True
+
+
+    def _load_to_canvas(path):
```

Comment:
We also need to convert this (dataobjects is gone, isn't it?). Will you, @NicolaiThomassen?

Reply:
dataobjects still exists. I had an other issue with it.

Reply:
ah, cool then.